xml/obs_build_constraints.xml

```diff
@@ -3,43 +3,125 @@
  xmlns:xlink="http://www.w3.org/1999/xlink" >
- <title>Build Job Constraints</title>
+ <title>Build Constraints</title>
  <info/>
- <para> Build job constraints can define requirements for the hardware or
-  software of the build host. Constraints can be defined per package or for
-  repositories. </para>
- <para> The build constraints for an entire project or specific repositories is
-  part of the project config. For each constraint, it contains a line</para>
-  <screen>Constraint: &lt;SELECTOR&gt; &lt;STRING&gt;</screen> 
-  <para>The selector is a colon-separated list. </para>
- <para> The build constraints for a package are part of the package sources, as
-  a _constraints XML source file (validated on submission). The _constraints
-  source file can contain the values listed below. </para>
- <para>In addition to that some build recipe parsers support to parse constraints
-  also from the build descriptions, eg. in spec files. These can also be used
-  conditionally in their context. For example to apply these only on some architectures
-  or only for specific multibuild flavors.</para>
- <para>Constraints defined in project build config, _constraints and the build recipe are merged
-  in this order. This means a constraint for the same topic overwrites the one from the former location.</para>
- <para>NOTE: If no build host meets the constraints, the package will turn to a failed state.</para>
+ <para> Using build constraints, one can define hardware and/or software
+  requirements that the Build Service will then use to narrow down the pool
+  of build workers that are qualified to build a particular project, package or
+  repository.
+ </para>
+ <para> By setting appropriate build constraints, one can avoid situations
+  in which builds fail due to insufficient memory or disk space, wrong kernel
+  version, etc. </para>
  <sect1>
-  <title><literal>hostlabel</literal></title>
-  <para> The hostlabel is any string which can be assigned to build hosts when
-   starting the bs_worker process. It can be used to run on specific hosts,
-   which may be used for running benchmarks in a reproducible way. This
-   constraint can also be defined as a negative definition using the
-   <literal>exclude=true</literal>
-   attribute. However, the hostlabel is always specific to one OBS instance.
-   You should avoid it as much as possible, since building with this constraint
-   in another instance is usually not possible. Use any of the other
-   constraints if possible. </para>
-  <para> Example for _constraints file:</para>
-   <screen>&lt;constraints exclude="false"&gt;
-  &lt;hostlabel&gt;benchmark_runner&lt;/hostlabel&gt;
+  <title>Setting constraints</title>
+  <para> Constraints can be set for a particular package by including a
+   _constraints file in the package. </para>
+  <para> Constraints can also be included directly in the build recipe (spec,
+   dsc, etc.), provided the build recipe parser supports it. </para>
```

Comment:
It would be nice if you would name the ones we *know* support it right now...

Reply:
it is written there already: "Example for rpm spec file or Dockerfile:"